### DISCUSSION TITLE
Mail referring to localhost instead of domain with docker-compose

### DISCUSSION THREAD

**Discussion by eddydc** [lukevella/rallly#159] [Q&A] [answered]
Hi,

Ive installed Rallly with docker-compose and set the .env file. A nginx proxy is uesed for the outside world. 
Mails are still referring to localhost:3000. Where/how can I set this to the host I've set with nginx?

Many thx in advance,

Eddy

**Comment by lukevella** [comment, marked as answer]
Hi, currently the domain is inferred from the headers of the request that triggers the email. Seems to work fine for me so not sure if there's anything specific about your configuration that is preventing it from working. Can you confirm that when you make the request you are using the domain specified in your nginx config?

**Reply by eddydc** [reply]
In the mail that asks to verify e-mail, all the links provided are referring to http://localhost:3333 (I had to choose another port since there is other software on my VPS using port 3000) instead of the HTTPS (sub)domain I've set up in nginx that's proxying to localhost:3333 
In a sense I can understand that the mail is wrong since the docker container doesn't know anything about the domain I'm using since nothing is set in the .env
Is there an option to do so? If not, how can I make the container believe that it needs to use my domain instead of localhost:3333

**Reply by lukevella** [reply]
> In a sense I can understand that the mail is wrong since the docker container doesn't know anything about the domain I'm using since nothing is set in the .env

[See here](https://github.com/lukevella/rallly/blob/main/utils/absolute-url.ts). When a request is received by the container it will get the domain from the request header. Your reverse proxy should be setting a header with the domain from which the request originated. If this doesn't work then your reverse proxy is not correctly configured. I'm not the right person to ask for nginx help but you can google how to get nginx to forward the host name. It should be fairly straight forward, think you just need to add this line to your config.

```
proxy_set_header X-Forwarded-Host $host;
```

That said, I will consider adding an env variable to override this in the future. I was just trying to keep the config as lean as possible.

**Reply by eddydc** [reply]
Oops, my fault. I do normally set this in each nginx proxy config, but for some reason I forget to uncomment this and handled with the assumption I did and looked at the wrong location. My sincere apologies for this.
Having said this, an option in the .env would be nice too and maybe some setting in order to set the L&F, language and the home page are also nice to haves. Or did I miss something here too. 
Many thanks again for the quick help.

**Reply by lukevella** [reply]
> Having said this, an option in the .env would be nice too and maybe some setting in order to set the L&F, language and the home page are also nice to haves. Or did I miss something here too.

Don't think you missed anything. Will work on making it more configurable at some point. I'm just focusing on other things which are higher priority at the moment.
